docs/modules/ROOT/pages/adr/0043-strimzi-operator-for-kafka.adoc

```diff
@@ -0,0 +1,446 @@
+
+* Use Crossplane Provider Kubernetes to create Strimzi `Kafka` CRs
+* Use Crossplane compositions to abstract Kafka cluster configuration
+* Configure proper resource requests and limits based on expected load
+* Implement topologySpreadConstraints and Kafka Rack awareness for multi-AZ distribution
+* Enable Prometheus ServiceMonitors / PodMonitors for all components
+* Deploy with KRaft mode (ZooKeeper-less)
+
+=== Client Configuration
+
+* Provide starter templates for Java applications (Quarkus, Spring Boot)
+* Include sample KafkaUser and KafkaTopic resources that can be used with the cluster
+* Document best practices for consumer lag monitoring
+* Provide guidance on replication factor and min.insync.replicas configuration
+
+
+=== Backup and Recovery
+
+Backup and recovery of Kafka require careful consideration because Kafka is a distributed system where durability is primarily provided by replication. The platform objective is to provide a reliable emergency restore capability while encouraging application teams to adopt patterns that reduce reliance on point-in-time restores (idempotent processing, event sourcing, tiered retention, compacted topics, etc.).
+
+Scope and Responsibility::
+*Platform*: provide PV snapshot capability for Kafka PVCs, configuration for snapshot retention/rotation, a documented emergency restore playbook, and tools for cluster-level restores. The platform will also snapshot Schema Registry storage and configuration where possible.
+*Application Teams*: own application-level recovery requirements (single-topic or message-level restores), define RPO/RTO needs, and adopt application patterns that avoid the need for frequent restores.
+
+Snapshot Frequency & RPO Options::
+* **Daily snapshots (default)** — RPO ≈ 24 hours. Reasonable for most use cases; minimal storage cost.
+* **Hourly snapshots (optional)** — RPO ≈ 1 hour. Enable for mission-critical clusters where acceptable.
+
+
+RTO Expectations (high level)::
+* **Full cluster restore**: dependent on cluster size and I/O performance — expect several hours for small clusters, longer for larger clusters. Restoration includes PV restore time, broker startup, partition leader elections and potential rebalancing.
+* **Single-topic / message restores**: may require custom tooling or manual processes; expect non-trivial effort and longer RTOs.
+
+
+Notes on Single-Topic / Message-Level Recovery::
+* This is not natively supported by PV snapshots. Typical approaches include:
+   - Restoring an entire broker state and extracting topic data via consumer tooling (time-consuming).
+   - Using MirrorMaker/replication clusters to copy topic data before destructive maintenance (proactive strategy).
+   - Exporting topic data into an external store for ad-hoc restore (requires per-application design).
+* These approaches have trade-offs in cost, complexity and RTO; platform-level support is limited to emergency workflows and guidance.
+
+
+Limitations and Caveats::
```

Comment:
IMHO a pretty important caveat:

Snapshots can't protect against complete loss of the underlying cluster/storage. Some CSPs support off-site snapshots, however that's usually a feature of the big ones (AWS,GCP and Azure)

Reply:
I think k8up will do as well, thanks for the hint. As we can have huge amounts of data going though Kafka we need to be able to tune it on a per installation base I guess.